### PR TITLE
Revert "Revert "Implement scheduled event covers""

### PR DESCRIPTION
**Pull request by ooliver1**
Reverts nextcord/nextcord#413

### CODE REVIEW

File: nextcord/asset.py [hunk 1]
```diff
@@ -251,6 +251,14 @@ def _from_user_banner(cls, state, user_id: int, banner_hash: str) -> Asset:
             key=banner_hash,
             animated=animated
         )
+    @classmethod
+    def _from_scheduled_event_cover(cls, state, event_id: int, cover_hash: str) -> Asset:
```
Comment (TAG-Epic) [inline nextcord/asset.py:255]:
```suggestion
    def _from_scheduled_event_image(cls, state, event_id: int, cover_hash: str) -> Asset:
```

Comment (TAG-Epic) [inline nextcord/asset.py:255]:
```suggestion
    def _from_scheduled_event_cover(cls, state, event_id: int, image_hash: str) -> Asset:
```

File: nextcord/asset.py [hunk 2]
```diff
@@ -251,6 +251,14 @@ def _from_user_banner(cls, state, user_id: int, banner_hash: str) -> Asset:
...
+        return cls(
+            state,
+            url=f'{cls.BASE}/guild-events/{event_id}/{cover_hash}.png',
```
Comment (TAG-Epic) [inline nextcord/asset.py:258]:
```suggestion
            url=f'{cls.BASE}/guild-events/{event_id}/{image_hash}.png',
```

File: nextcord/scheduled_events.py [hunk 3]
```diff
@@ -204,8 +204,10 @@ class ScheduledEvent(Hashable):
         The scheduled start time for the event.
     user_count: :class:`int`
         An approximate count of the 'interested' users.
+    cover: :class:`Asset`
```
Comment (TAG-Epic) [inline nextcord/scheduled_events.py:207]:
```suggestion
    image: :class:`Asset`
```

File: nextcord/scheduled_events.py [hunk 4]
```diff
@@ -220,6 +222,7 @@ class ScheduledEvent(Hashable):
         'user_count',
         '_state',
         '_users',
+        'cover',
```
Comment (TAG-Epic) [inline nextcord/scheduled_events.py:225]:
```suggestion
        'image',
```

File: nextcord/scheduled_events.py [hunk 5]
```diff
@@ -250,6 +253,11 @@ def _update(self, data: ScheduledEventPayload) -> None:
         self.channel_id: Optional[int] = data.get('channel_id')
         self._users: Dict[int, ScheduledEventUser] = {}
         self._update_users(data.get('users', []))
+        
+        if cover := data.get("image"):
```
Comment (TAG-Epic) [inline nextcord/scheduled_events.py:257]:
```suggestion
        if image := data.get("image"):
```

File: nextcord/scheduled_events.py [hunk 6]
```diff
@@ -250,6 +253,11 @@ def _update(self, data: ScheduledEventPayload) -> None:
...
+            self.cover: Optional[Asset] = Asset._from_scheduled_event_cover(self._state, self.id, cover)
```
Comment (TAG-Epic) [inline nextcord/scheduled_events.py:258]:
```suggestion
            self.image: Optional[Asset] = Asset._from_scheduled_event_image(self._state, self.id, image)
```

File: nextcord/scheduled_events.py [hunk 7]
```diff
@@ -250,6 +253,11 @@ def _update(self, data: ScheduledEventPayload) -> None:
...
+        else:
+            self.cover: Optional[Asset] = None
```
Comment (TAG-Epic) [inline nextcord/scheduled_events.py:260]:
```suggestion
            self.image: Optional[Asset] = None
```

**Review by TAG-Epic — CHANGES_REQUESTED** [review]

**Comment by ooliver1** [comment]
absolute pain as i cant request changes for self

**Pull request merged**